internal/verifier/verifier.go

```diff
@@ -172,13 +172,13 @@ func (verifier *ProvenanceIRVerifier) Verify() (VerificationResult, error) {
 	}
 
 	// Verify HasBuildCmd.
-	if verifier.Want.WantBuildCmds {
+	if verifier.Got.HasBuildCmd() && verifier.Want.WantBuildCmds {
```

Comment:
Just checking: "Does not have" is different from "cannot have". For instance, if the BuildType is Amber and the command is missing, we want this check to fail. Unless the conversion from Amber to ProvenanceIR checks that a command is present. Is that the case?

Reply:
You are absolutely right! 

So how I have achieved the distinction between "does not have" and "cannot have":

I have made every optional field in `ProvenanceIR` a pointer.

If `NewProvenance` is called `WithX`, then this pointer is set (possibly to the default value). 

`WithX` is called in `FromAmber` / `FromSlsav02`... So we have to maintain the invariant that `FromP` sets every value P _can_ have via `WithX`. 

Then `HasX` equals `false` (is `nil), this means that the provenance cannot have the value. 

If, e.g., for `build_cmd` we have an empty list (not `nil`), then the provenance does not have the value.

Reply:
Ok, great. That sounds good. Thanks :)